xapian-core/api/omenquireinternal.h

```diff
@@ -313,6 +313,21 @@ class MSet::Internal : public Xapian::Internal::intrusive_base {
 	/** Fetch items specified into the document cache.
 	 */
 	void fetch_items(Xapian::doccount first, Xapian::doccount last) const;
+
+	/** Updates the weight corresponding to the document indexed at position i with wt_.
+	 *  Updates the max_possible and max_attained for the MSet.
+	 *  set_item_weight must be called upon to update the weight of every document in MSet
+	 *  in the same order as that of the documents present in MSet to avoid miscalculation of
+	 *  max_attained and max_possible.
```

Comment:
There seems to be a word missing after "must be called upon".

I'm not clear why the "same order" part would make any difference here.